### DISCUSSION TITLE
Performance Degradation over time: Longer duration flights > 3 hours

### DISCUSSION THREAD

**Discussion by deepan-mehta-analytics** [flybywiresim/aircraft#5512] [Help] [answered]
After SU5 I've been using the dev version and keeping it updated via the installer.

Have used the suggestions to fly not using live weather.

Whats happening now on the longer duration flights, is that initially on departure the performance is just fine 40- 50+ FPS depending on the airport. 

However by the time the aircraft is at TOD , the frames drop by  25-30FPS.  The final FPS is sometimes 17 to 20.

PC Config 3900X @4.3 , 64GB ram , 2080TI
Most settings on ULTRA , render scaling 145 on a 1080p monitor ( tried with 100 too ) 

MSFS dev mode shows that the main thread performance massively degrades , while starting the flight dev mode all greens, end of flight main thread takes it to all reds.

any similar experiences ? suggestions ,  Thanks in advance

**Comment by deepan-mehta-analytics** [comment, marked as answer]
In case anyone faces that issue:
After trying several setting changes, seems to be a "online service" issue after SU5. Switched off "online service" the performance gets restored.....will await the ASOBO fixes/update

**Reply by mbochove** [reply]
Thanks for this, I had the same issue yesterday on a 4,5hr flight. Do you mean the "online functionality" option? And did you switch it off and on again? Or did you leave it set to off?

**Reply by deepan-mehta-analytics** [reply]
Hi yes the "online functionality" option, so in my case it has been happening progressively towards the end of the flight. 

I waited till I saw the FPS modes go red in the MSFS dev mode display then I switched it off and then back on again immediately and the FPS were restored.

However I have to say its all very tricky , I got a CTD while trying it once, but thankfully I had completed the flight  .

I haven't tried to complete the flight by leaving the online functionality off because I didn't want to loose the visual quality in any way. I'll try it on the next flight though...leaving it off at the point the fps start to drop.

Do post if you learn more about it...Thanks

**Reply by mbochove** [reply]
Thanks for the reply, I'll give it a go the next time I experience it. I noticed the gradual degradation as well, and the following things are maybe worth noting:

- I started the flight around 45fps and ended with 15fps
- Usually if you move the camera all the way down or up in the cockpit, the fps will go up since there is no "outside" to be rendered. In my case it stuck to 15fps
- Even changing all my settings to low (I normally play ultra) makes the fps stick to 15
- My GPU and CPU were near idle, so it seems it's not really addressing the resources
- I finished my flight and exited MSFS, then started it again, and everything was fine

Interesting stuff!

**Comment by mbochove** [comment]
I wanted to come back to this as i've had the gradual decline in FPS again. Indeed disabling the "online functionality" and turning it on again works (although it takes about a minute or so to come back up to normal FPS).

But... its not A32nx related. I had it this time flying the salty 747 mod. So hopefully if someone finds it here, you're lucky, but its rather something Asobo will have to fix :)

**Reply by deepan-mehta-analytics** [reply]
Thanks for the updated info, yes I'd also concluded that its not a A32NX issue earlier on.

Data loading from the servers definitely seems to be the issue, one other indicator of this is that I often get a "bandwidth too low " for photogrammetry error. The sim then puts out a pop-up giving the option to "ignore" or "turn off photogrammetry".

I have a decent enough connection which normally runs at 256 Mbps fulfilling the stated requirements. Download speeds for core MSFS update files run around 20 Mbps.

So it seems that the data stream seems to be "Throttling" performance at the PC end, laymans words but I'm not even sure we can use that term because GPU and CPU usage also drops when the frames drop to the low numbers. ( You had observed earlier too)

My latest workaround involves using the Nvidia 3D settings  via control panel.  I've set VSync to using "FAST" mode there and OFF in the MSFS settings..... thats been keeping the frames to at least 30 FPS at the drop time ( last phases of flight) ....start FPS are 40 to 60

So all indicators of it being server data load issues, I still do switch the "online functionality" on and off  and the interesting bit to note is that it doesn't set the FPS to start of flight rates

Hoping SU6 will solve it.....enjoying the fabulous A32NX project either way :-)

**Comment by deepan-mehta-analytics** [comment]
![msfs](https://user-images.githubusercontent.com/83442181/132941673-ec0ae5c2-88d3-4be2-9f74-19cf9456672a.png)

After SU6 , GPU usage down to running at 1350 MHz  as against 1950 MHz normally , CPU Core 0 only max 8-10 % 

FPS down to 17-21 from initial 40+ regular and 50+ on discovery flights

Screen Shot @ 1080p because of the render scaling at 100 , normally have it at 145

Enjoying the sim anyway :-)
